### PR TITLE
Create #4066 - Theme-controlled color scheme switch.md

### PR DESCRIPTION
**Pull request by arkthur**
## Summary of the Pull Request
Specs for feature request "Theme-controlled color scheme switch".

 
## References
#4066

### CODE REVIEW

**Comment by zadjii-msft** [comment]
Thanks for the PR! I'm gonna tag this up so we discuss it during our next team sync next Monday.

**Comment by arkthur** [comment]
> Thanks for the PR! I'm gonna tag this up so we discuss it during our next team sync next Monday.

Any news about this? I'm around in case anything is missing ;)

**Comment by zadjii-msft** [comment]
Team sync usually isn't till later in the day on Mondays usually 😉 That being said, we've got a bunch of OOFages right now so I think we're gonna have to punt that meeting back a few days. Sorry for the delays 😬

**Comment by arkthur** [comment]
I understand. I hope that meeting happens sooner than expected. In any case, I'll keep coming back in case more information is needed.

Thanks for the help until now :D

**Comment by zadjii-msft** [comment]
Okay I owe you a longer reply in the morning, but the team was generally on board with this and happy with the spec. We've got tiny nits of things we might have added if we wrote it ourselves, but overall we're happy. I'll come back with those in the morning.

File: doc/specs/#4066 - Theme-controlled color scheme switch.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+## Solution Design
+
+Could be implemented in the form of:
+
+```json
+"colorScheme": {
+    "light": "BlulocoLight",
+    "dark": "BlulocoDark"
+}
+```
```
Comment (zadjii-msft) [inline doc/specs/#4066 - Theme-controlled color scheme switch.md:33]: Team consensus was that both were good, but we were drawn to this implementation more. We could encapsulate the color scheme in the profile as something like a `ColorSchemeEntries` class, or heck even just a winrt `IMap<hstring, hstring>`, and look up the one we needed as necessary.

File: doc/specs/#4066 - Theme-controlled color scheme switch.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+or:
+
+```json
+    "colorSchemeLight": "BlulocoLight",
+    "colorSchemeDark": "BlulocoDark"
+```
+
```
Comment (zadjii-msft) [inline doc/specs/#4066 - Theme-controlled color scheme switch.md:40]: We may want to add a note about what happens if the user saves the settings with just `"colorScheme": "Campbell"` set. Does that roundtrip as 

```json
"colorScheme": "Campbell"
```

or 

```json
"colorScheme": {
    "light": "Campbell",
    "dark": "Campbell"
}
```

?

File: doc/specs/#4066 - Theme-controlled color scheme switch.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## UI/UX Design
+
+In a first version it could look like the terminal in Visual Studio Code, and an improvement could be to have light mode specific color schemes, just like those already present in Windows terminal. A good idea could be to get an inspiration in Dark++ and Light++ VSCode color themes.
+
+A user could benefit from a more healthy light level contrast between the screen their looking at and the environment they are, reducing the risk of headache or developing/intensifying eye problems, and any other related eye conditions. Plus, it adds to a more consistent experience between different programs and apps, and the system itself.
+
```
Comment (zadjii-msft) [inline doc/specs/#4066 - Theme-controlled color scheme switch.md:46]: (if this were us)

We might add a section here about what happens in the Settings UI if a user does enable this feature. For now though, we're cool with just replacing the color scheme dropdown on the Profile Appearance page with a text box that says "this setting must be configured in the settings file", if the scheme is set to a set of different schemes. 

We also have half a mind to adjust how schemes are configured via the UI entirely, so we're not gonna ask you to add this bit 😝 This is firmly an "us" problem.

File: doc/specs/#4066 - Theme-controlled color scheme switch.md (hunk 2 above)
Comment (zadjii-msft) [inline doc/specs/#4066 - Theme-controlled color scheme switch.md:40]: Another note: we actually already have the eventing set up in the Terminal for listening to theme change notifications from the OS, so once the logic for pulling the correct scheme out of the profile based on the OS theme is set up, the rest of the plumbing _should just work_

**Review by zadjii-msft — APPROVED** [review]
Okay, nits scattered throughout. Overall, we're jazzed to see a spec for this. We're also cautiously optimistic that implementation really shouldn't be all that difficult.

(we're also in a bit of a fire drill internally, and likely will be for a few weeks still, so sorry if this languishes a bit)

File: doc/specs/#4066 - Theme-controlled color scheme switch.md (hunk 1 above)
Comment (arkthur) [reply doc/specs/#4066 - Theme-controlled color scheme switch.md:33]: I agree. I like this one better too ;)

File: doc/specs/#4066 - Theme-controlled color scheme switch.md (hunk 2 above)
Comment (arkthur) [reply doc/specs/#4066 - Theme-controlled color scheme switch.md:40]: I think the second one would be more standard and would serve as a good starting point for users to get used to the new way of setting color schemes.

File: doc/specs/#4066 - Theme-controlled color scheme switch.md (hunk 3 above)
Comment (arkthur) [reply doc/specs/#4066 - Theme-controlled color scheme switch.md:46]: I appreciate that last part 😆

Although, I'm on board with all the "this setting must be configured in the settings file" thingy. I think it's wise to introduce such a change as gradually as possible.

File: doc/specs/#4066 - Theme-controlled color scheme switch.md (hunk 2 above)
Comment (arkthur) [reply doc/specs/#4066 - Theme-controlled color scheme switch.md:40]: Just as I imagined it. Perfect!

Comment (miniksa) [reply doc/specs/#4066 - Theme-controlled color scheme switch.md:40]: Agree with @arkthur. I would have the parser accept the old format but write the new one where it fills it into both light and dark.

**Review by miniksa — APPROVED** [review]
This looks great to me!

**Review by raghavthind2005 — APPROVED** [review]

**Pull request merged**